thefuck/rules/git_push.py

```diff
@@ -39,6 +39,6 @@ def get_new_command(command):
         while len(command_parts) > push_idx and command_parts[len(command_parts) - 1][0] != '-':
             command_parts.pop(len(command_parts) - 1)
 
-    arguments = re.findall(r'git push (.*)', command.output)[0].strip()
+    arguments = re.findall(r'git push (.*)', command.output)[0].replace("'", r"\'").strip()
```

Comment:
Instead of replacing just single quotes, I think we should use [`shell.quote`](https://github.com/nvbn/thefuck/blob/83e1710712a9b48fee1822109a10a8324fe64930/thefuck/shells/generic.py#L101-L109) here.

Reply:
shell.quote uses shlex.quote which escapes the whole string; so you end up with something like:
`'--set-upstream origin test'"'"'s'`

Reply:
Oh, I had overlooked the fact that the command contains other arguments as well. Thanks for pointing that out. This seems fine to me now.

Reply:
Yes, I had hoped there was something better, and I had a quick search, but nothing jumped out to me.  I do agree though, that this is kind of a sticking-plaster.  I do find it weird that Git allows you to use quotes in branch names...